src/db.c

```diff
@@ -1536,12 +1536,12 @@ int expireIfNeeded(redisDb *db, robj *key) {
     if (checkClientPauseTimeoutAndReturnIfPaused()) return 1;
 
     /* Delete the key */
+    int retval = server.lazyfree_lazy_expire ? dbAsyncDelete(db,key) :
+                                               dbSyncDelete(db,key);
     server.stat_expiredkeys++;
     propagateExpire(db,key,server.lazyfree_lazy_expire);
     notifyKeyspaceEvent(NOTIFY_EXPIRED,
         "expired",key,db->id);
-    int retval = server.lazyfree_lazy_expire ? dbAsyncDelete(db,key) :
-                                               dbSyncDelete(db,key);
     if (retval) signalModifiedKey(NULL,db,key);
```

Comment:
general question: how come `retval` is test only for `signalModifiedKey`? shouldn't it affect `notifyKeyspaceEvent` and others

Reply:
@oranagra ^

Reply:
this seems to be dead code.
since `keyIsExpired` returns positive, we can conclude the key exists, and dbSyncDelete / dbAsyncDelete will always return 1.
looks like an innocent change added in 20eeddfb8a8840aff8ac9e54a69650890d9c1e64
we should either let it skip the rest of the function (propagateExpire, notifyKeyspaceEvent etc) by using an early exit, or just remove this `retval` variable and always return 1.

@ShooterIT care to enhance your PR with these changes?

Reply:
> since keyIsExpired returns positive, we can conclude the key exists, and dbSyncDelete / dbAsyncDelete will always return 1.

According to this, i prefer to remove `retval` variable, WDYT @oranagra @guybe7

Reply:
i agree... would make the code less confusing.